src/main/java/mekanism/common/block/BlockBasic.java

```diff
@@ -412,12 +414,27 @@ public void onBlockClicked(World world, BlockPos pos, EntityPlayer player)
 			{
 				if(!bin.bottomStack.isEmpty())
 				{
-					if(!player.isSneaking())
+					ItemStack stack;
+					if(player.isSneaking())
 					{
-						world.spawnEntity(new EntityItem(world, player.posX, player.posY, player.posZ, bin.removeStack().copy()));
+						stack = bin.remove(1).copy();
 					}
-					else {
-						world.spawnEntity(new EntityItem(world, player.posX, player.posY, player.posZ, bin.remove(1).copy()));
+					else
+					{
+						stack = bin.removeStack().copy();
+					}
+
+					if(!player.inventory.addItemStackToInventory(stack))
+					{
+						BlockPos dropPos = pos.offset(bin.facing);
```

Comment:
hmm I guess it won't make a huge difference since you have to be within reaching distance of the face anyway.

Minor difference of 'oops I can't pick that up' vs 'oops my hands & inventory is full (but I can pick it up for a split second?)' lol

Also I can't remember if you can activate this in creative, but if you can the `addItemStackToInventory` will never fail, even if it is full